synthesizer/process/src/stack/register_types/mod.rs

```diff
@@ -98,6 +98,7 @@ impl<N: Network> RegisterTypes<N> {
                 RegisterType::Plaintext(PlaintextType::Literal(LiteralType::Address))
             }
             Operand::BlockHeight => bail!("'block.height' is not a valid operand in a non-finalize context."),
+            Operand::NetworkID => bail!("'network.id' is not a valid operand in a non-finalize context."),
```

Comment:
Should it be valid?

Reply:
Discussed offline. Excluding in execution and evaluation does not prevent the desired use case as you can directly pass in a u32 and check it against the network in a finalize block.